src/lib/api/core/fetcher.ts

```diff
@@ -0,0 +1,60 @@
+import { API_CONFIG } from '../config';
+
+class ApiFetcher {
+  private baseurl: string;
+
+  constructor(baseUrl: string) {
+    this.baseurl = baseUrl;
+  }
+
+  private handleResponse = async (response: Response) => {
+    try {
+      const data = await response.json();
+      return data;
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  private getHeaders() {
+    return {
+      'Content-type': 'application/json',
+    };
+  }
+
+  async get<T>(path: string): Promise<T> {
+    const response = await fetch(`${this.baseurl}/${path}`, {
+      headers: this.getHeaders(),
+    });
+    return this.handleResponse(response);
+  }
```

Comment:
try {
   const response = await fetch(`${this.baseurl}/${path}`, {
      headers: this.getHeaders(),
    });
      const data = await response.json();
      return data;
    } catch (error) {
      throw error
    }